src/full_node/block_store.py

```diff
@@ -133,11 +136,119 @@ async def get_sub_block_records(
         for row in rows:
             header_hash = bytes.fromhex(row[0])
             ret[header_hash] = SubBlockRecord.from_bytes(row[3])
-            if row[4]:
+            if row[5]:
                 assert peak is None  # Sanity check, only one peak
                 peak = header_hash
         return ret, peak
 
+    async def get_headers_in_range(
+        self,
+        start: int,
+        stop: int,
+    ) -> Dict[bytes32, HeaderBlock]:
+
+        formatted_str = (
+            f"SELECT header_hash,block from full_blocks WHERE sub_height >= {start} and sub_height <= {stop}"
+        )
+
+        cursor = await self.db.execute(formatted_str)
+        rows = await cursor.fetchall()
+        await cursor.close()
+        ret: Dict[bytes32, HeaderBlock] = {}
+        for row in rows:
+            header_hash = bytes.fromhex(row[0])
+            full_block: FullBlock = FullBlock.from_bytes(row[1])
+            ret[header_hash] = full_block.get_block_header()
+
+        return ret
+
+    async def get_sub_block_in_range(
```

Comment:
Rename to get_sub_block_records_in_range?

Reply:
sure, we are using typing though so there isn't a real need to state the type in the name